scripts/zones/Southern_San_dOria/npcs/Kipopo.lua

```diff
@@ -6,18 +6,41 @@
 -----------------------------------
 require("scripts/globals/status");
 require("scripts/globals/crafting");
+require("scripts/globals/quests");
 local ID = require("scripts/zones/Southern_San_dOria/IDs");
 -----------------------------------
 
 function onTrade(player,npc,trade)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
+
+    if (player:hasKeyItem(dsp.ki.TORN_PATCHES_OF_LEATHER) and sayItWithAHandbagCS == 2 and trade:getItemCount() == 3) then
+        if (trade:hasItemQty(2012,1) and trade:hasItemQty(850,1) and trade:hasItemQty(816,1)) then
+        player:startEvent(910)
+        end
+    end
 end;
 
 function onTrigger(player,npc)
+    local sayItWithAHandbag = player:getQuestStatus(CRYSTAL_WAR, dsp.quest.id.crystalWar.SAY_IT_WITH_A_HANDBAG)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
     local guildMember = isGuildMember(player,7);
     local SkillCap = getCraftSkillCap(player, dsp.skill.LEATHERCRAFT);
     local SkillLevel = player:getSkillLevel(dsp.skill.LEATHERCRAFT);
 
-    if (guildMember == 1) then
+    if (sayItWithAHandbag == QUEST_COMPLETED and sayItWithAHandbagCS == 4 and player:hasItem(19110)) then
+        player:startEvent(914)
+        player:setVar("sayItWithAHandbagCS", 5)
+    elseif (player:hasKeyItem(dsp.ki.REPAIRED_HANDBAG) and sayItWithAHandbagCS == 4) then
+        player:startEvent(913)
+    elseif (sayItWithAHandbagCS == 3 and player:needToZone() == false) then
+        player:startEvent(912)
+    elseif (sayItWithAHandbagCS == 3 and player:needToZone() == true) then
```

Comment:
Could also:
```
if sayItWithAHandbagCS == 3 then
    if player:needToZone() then
        player:startEvent(911)
    else
        player:startEvent(912)
    end
end
```

Reply:
will try to remember this one, you are right as they are evaluating the same CS 😸